pkgs/fc/agent/default.nix

```diff
@@ -34,6 +35,20 @@ let
     buildInputs = [ pytest structlog ];
   };
 
+  stamina = py.buildPythonPackage rec {
+    pname = "stamina";
+    version = "23.1.0";
+    format = "pyproject";
+
+    src = fetchPypi {
+      inherit pname version;
+      hash = "sha256-sWzj1S1liqdduBP8amZht3Cr/qkV9yzaSOMl8qeFR4Y=";
+    };
+
+    nativeBuildInputs = with py; [ hatchling hatch-vcs hatch-fancy-pypi-readme ];
+    propagatedBuildInputs = with py; [ structlog tenacity typing-extensions ];
```

Comment:
I think it's ok here because `with py;` is close to the place where the package is used and adding another explicit `py.tenacity` looks strange . In general, qualifying packages is a good idea when packages from multiple sources are mixed in a list.